Appsilon/rhino
# Allow custom `RHINO_NPM` command (support `bun` and `pnpm`)

### Changes

This PR is a copy of #496, rebased onto `main` and with review suggestions applied. I couldn't edit the PR directly as it's in a forked repository; @nbbn is on sick leave till the end of the week and we don't want to wait (release 1.6 is underway).

### How to test

1. Install `npm` alternative of your choice, e.g. `bun` or `pnpm`. Alternatively, create a simple `wrapped-npm` script (make it executable and place in your `PATH`):
    ```sh
    #!/bin/sh
    echo "Wrapper!"
    exec npm "$@"
    ```
2. Run e.g. `rhino::lint_js()` in a Rhino project. It should use `npm` by default.
3. Set `RHINO_NPM` to the command of your choice and try `rhino::lint_js()` again. You can do it by running R like this: `RHINO_NPM=bun R`.
